src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1343,25 +1372,41 @@ void uct_dc_mlx5_ep_pending_purge(uct_ep_h tl_ep, uct_pending_purge_callback_t c
     }
 }
 
-ucs_status_t uct_dc_mlx5_ep_check_fc(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
+ucs_status_t uct_dc_mlx5_ep_check_fc(uct_dc_mlx5_iface_t *iface,
+                                     uct_dc_mlx5_ep_t *ep)
 {
+    ucs_status_t ret_status = UCS_OK;
     ucs_status_t status;
 
     if (iface->super.super.config.fc_enabled) {
-        UCT_RC_CHECK_FC_WND(&ep->fc, ep->super.stats);
-        if (ep->fc.fc_wnd == iface->super.super.config.fc_hard_thresh) {
+        if (ucs_unlikely((ep->fc.fc_wnd ==
+                          iface->super.super.config.fc_hard_thresh) ||
+                         (ep->fc.fc_wnd <= 0))) {
+            if (ep->fc.fc_wnd <= 0) {
+                ret_status = UCS_ERR_NO_RESOURCE;
+            }
+
             status = uct_rc_fc_ctrl(&ep->super.super,
                                     UCT_RC_EP_FLAG_FC_HARD_REQ,
                                     NULL);
             if (status != UCS_OK) {
+                if ((ep->dci != UCT_DC_MLX5_EP_NO_DCI) && 
+                    !uct_dc_mlx5_iface_is_dci_rand(iface)) {
+                    ucs_assertv_always(uct_dc_mlx5_iface_dci_has_outstanding(
```

Comment:
convert to regular assert, move to end of function (to check for any return value)

Reply:
done